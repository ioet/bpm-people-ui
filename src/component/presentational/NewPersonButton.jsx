import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button/Button';
import { withStyles } from '@material-ui/core';
import { NewPersonButtonStyles } from '../../styles';
import { NewPersonButtonConst } from '../../constants';

const NewPersonButton = (props) => {
  const {
    classes, createUser,
  } = props;

  return (
    <Button
      className={classes.button}
      variant="contained"
      color="secondary"
      onClick={(e) => {
        e.preventDefault();
        createUser();
      }}
    >
      {NewPersonButtonConst.ADD_NEW_PERSON}
    </Button>
  );
};

NewPersonButton.propTypes = {
  classes: PropTypes.object.isRequired,
  createUser: PropTypes.func.isRequired,
};

export default withStyles(NewPersonButtonStyles)(NewPersonButton);
