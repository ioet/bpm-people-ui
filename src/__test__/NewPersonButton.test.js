import React from 'react';
import NewPersonButton from '../component/presentational/BpmButton';

describe('<BpmButton />', () => {
  test('Component Renders correctly', () => {
    const appWrapper = shallow(<NewPersonButton />);

    expect(toJson(appWrapper))
      .toMatchSnapshot();
  });
});
