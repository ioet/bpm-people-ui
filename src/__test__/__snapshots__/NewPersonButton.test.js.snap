// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewPersonButton /> Component Renders correctly 1`] = `
<NewPersonButton
  classes={
    Object {
      "button": "NewPersonButton-button-1",
    }
  }
/>
`;
